### PR TITLE
fix: don't include chore PRs in changelog

### LINKED ISSUE #1493: 🐛 Bug: Don't include chore PRs in changelogs and release notes
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/create-typescript-app/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

Generated changelogs / release notes should include only the PRs relevant to the actual released version. Non-user-facing things like `chore(deps): ...` PRs shouldn't be shown.

### Actual

All merged commits/PRs get mentioned 😞. Example: https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/releases/tag/v0.13.0 

> ```plaintext
> chore(deps): update dependency @typescript-eslint/eslint-plugin to v7.4.0 by @renovate in https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/pull/262
> chore(deps): update dependency @typescript-eslint/parser to v7.4.0 by @renovate in https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/pull/263
> ```

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by michaelfaith**
## PR Checklist

- [x] Addresses an existing open issue: fixes #1493
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/create-typescript-app/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/create-typescript-app/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

This change adjusts the release-it config for both this repo, as well as what the create script generates, so that only three categories of changes are included in the changelog entries: feat, fix, and perf.  The idea being that these are the only three types of changes that are generally user-facing.

Tested locally with some dummy `chore` commits that I undid after.

Closes: #1493

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,8 +32,8 @@
 		"lint:spelling": "cspell \"**\" \".github/**/*\"",
 		"prepare": "husky",
 		"test": "vitest",
-		"test:create": "npx tsx script/create-test-e2e.ts",
-		"test:initialize": "npx tsx script/initialize-test-e2e.ts",
+		"test:create": "tsx script/create-test-e2e.ts",
+		"test:initialize": "tsx script/initialize-test-e2e.ts",
```
Comment (JoshuaKGoldberg) [inline package.json:36]: Unrelated changes?

**Review by JoshuaKGoldberg — APPROVED** [review]
LGTM, thanks so much! This has been bugging me for the longest time 😄...

File: .release-it.json [hunk 2]
```diff
@@ -12,7 +12,19 @@
...
+				{ "section": "Features", "type": "feat" },
+				{ "section": "Bug Fixes", "type": "fix" },
+				{ "section": "Performance Improvements", "type": "perf" },
+				{ "hidden": true, "type": "build" },
+				{ "hidden": true, "type": "chore" },
+				{ "hidden": true, "type": "ci" },
+				{ "hidden": true, "type": "docs" },
+				{ "hidden": true, "type": "refactor" },
+				{ "hidden": true, "type": "style" },
+				{ "hidden": true, "type": "test" }
```
Comment (JoshuaKGoldberg) [inline .release-it.json:26]: [Praise] I appreciate you alphabetizing these 😁

File: package.json (hunk 1 above)
Comment (JoshuaKGoldberg) [reply package.json]: fyi @michaelfaith I removed these changes before merging - but if you think they're good I'm interested in learning more

**Comment by JoshuaKGoldberg** [comment]
The migration test failure is tracked in #1811. I'm hoping that the whole system will be removed within the next 1-2 weeks anyway 🤞 (#1181).

**Comment by JoshuaKGoldberg** [comment]
@all-contributors please add @michaelfaith for code.

> 🤖 Beep boop! This comment was added automatically by [all-contributors-auto-action](https://github.com/marketplace/actions/all-contributors-auto-action).
> Not all contributions can be detected from Git & GitHub alone. Please comment any missing contribution types this bot missed.
> ...and of course, thank you for contributing! 💙

File: package.json (hunk 1 above)
Comment (michaelfaith) [reply package.json]: yeah, no worries. That was unrelated. I thought it might have been missed in some repo migration from npm to pnpm.  It seemed strange that everything used pnpm except for these two scripts, which didn't really need `npx` either, since tsx is installed locally.  Unless you're wanting to allow people to run those scripts without needing to install pnpm or any dependencies?  But yeah, it was unrelated to this change; just something I noticed, while testing the changes I made.

**Comment by rakleed** [comment]
@michaelfaith I still see this on the [Releases page](https://github.com/JoshuaKGoldberg/create-typescript-app/releases). But in the original issue, it was mentioned that such categories [should also be excluded from Releases](https://github.com/JoshuaKGoldberg/create-typescript-app/issues/1493#:~:text=All%20merged%20commits/PRs%20get%20mentioned%20%F0%9F%98%9E.%20Example%3A%20https%3A//github.com/JoshuaKGoldberg/eslint%2Dplugin%2Dpackage%2Djson/releases/tag/v0.13.0), not only from CHANGELOG.

Or am I missing something?

**Comment by michaelfaith** [comment]
> @michaelfaith I still see this on the [Releases page](https://github.com/JoshuaKGoldberg/create-typescript-app/releases). But in the original issue, it was mentioned that such categories [should also be excluded from Releases](https://github.com/JoshuaKGoldberg/create-typescript-app/issues/1493#:~:text=All%20merged%20commits/PRs%20get%20mentioned%20%F0%9F%98%9E.%20Example%3A%20https%3A//github.com/JoshuaKGoldberg/eslint%2Dplugin%2Dpackage%2Djson/releases/tag/v0.13.0), not only from CHANGELOG.
> 
> Or am I missing something?

The changes in this PR only impact what's added to the changelog.  That config is for the conventional changelog plugin.   I'll take a look and see if there's similar config for the gh release page.  Those are two different underlying tools.

**Comment by michaelfaith** [comment]
Yeah, looking at the docs for the gh config, there isn't the same level of awareness of the groups.  The convention commits commit format (e.g. feat: title) that's used to generate the changelog is something that's specific to conventional commits.  The gh release page just does a simple git diff and dumps everything into the page.  It doesn't have the same level of commit subject parsing.  It does allow you to pass in a custom function to generate the release page, so we could effectively re-create the same behavior as conventional changelog ourselves, if you think it's worth it. 

https://github.com/release-it/release-it/blob/main/docs/github-releases.md#function

**Comment by michaelfaith** [comment]
@JoshuaKGoldberg since this didn't solve both sides of the original issue, do you think it's worth re-opening it?

**Comment by JoshuaKGoldberg** [comment]
How about opening another issue? This one is getting kind of long to read through.
